plugin-server/src/main/ingestion-queues/batch-processing/each-batch-onevent.ts

```diff
@@ -59,15 +53,17 @@ export async function eachMessageAppsOnEventHandlers(
     }
 }
 
-export async function eachBatchAppsOnEventHandlers(
-    payload: EachBatchPayload,
-    queue: KafkaJSIngestionConsumer
-): Promise<void> {
+export async function eachBatchAppsOnEventHandlers(payload: Message[], consumer: IngestionConsumer): Promise<void> {
+    const hub = consumer.pluginsServer
+    if (!consumer.consumer?.consumer) {
```

Comment:
surely we could fit one more consumer here 

`!consumer.consumer?.consumer`is a great line of code 🤣

Reply:
![image](https://github.com/user-attachments/assets/fa0c0d6e-8e38-4e3f-af9f-882c6beb467b)